msg/ZFS-8000-ER/index.html

```diff
@@ -225,4 +225,91 @@
+<!-- end: templates/success -->
+<!-- start: article -->
+<!-- code: ZFS-8000-ER -->
+<!-- keys: errata -->
+<table width="80%"><tr><td>
+<a name="4"><b>ZFS Errata #4</b></a>
+<p>
+<dl>
+<dt><p><b>Type</b>
+<dd>Compatibility
+<dt><p><b>Severity</b>
+<dd>Moderate
+<dt><p><b>Description</b>
+<dd>An encrypted dataset contains an on-disk format incompatibility.
+<dt><p><b>Automated Response</b>
+<dd>No automated response will be taken.
+<dt><p><b>Impact</b>
+<dd>
+Encrypted datasets created before the ZFS packages were updated cannot be
+backed up via a raw send to an updated system. These datasets also cannot
+receive additional snapshots.
+</dd>
+<dt><p><b>Suggested Action for System Administrator</b>
+<dd>
+First, system administrators with affected pools will need to enable the
+<b>bookmark_v2</b> feature on their pools. This will disable read-write
+and read-only compatibility with previous versions of the ZFS software.
+Next, administrators will need to backup all existing encrypted datasets.
+This can be done in several ways. Non-raw <b>zfs send</b> and
+<b>zfs receive</b> can be used as per usual, as can traditional backup
+tools.
+
+If the administrator wishes to continue using raw send streams, the
+zfs_disable_ivset_guid_check tunable must be set to 1 on the receive side.
+Streams received this way (as well as any received before the upgrade)
+will need to be checked to ensure they are not corrupted.
+<pre>
+# zpool status
+  pool: test
+ state: ONLINE
+status: Errata #4 detected.
+        Existing encrypted datasets contain an on-disk incompatibility
+        which needs to be corrected.
+action: To correct the issue enable the bookmark_v2 feature and backup
+        existing encrypted datasets to new encrypted datasets and
+        destroy the old ones.
+   see: http://zfsonlinux.org/msg/ZFS-8000-ER
+  scan: none requested
+config:
+
+        NAME           STATE     READ WRITE CKSUM
+        test           ONLINE       0     0     0
+          /root/vdev0  ONLINE       0     0     0
+
+errors: No known data errors
+</pre>
+
+Import the pool and enable the <b>bookmark_v2</b> feature. Then backup any
+existing encrypted datasets to new datasets. This can be done with
+traditional tools or via <b>zfs send</b>. Raw sends will require that the
+zfs_disable_ivset_guid_check is set to 1 on the receive side. Once this
+is done, the original datasets should be destroyed.
+<pre>
+# zpool set feature@bookmark_v2=enabled test
+# echo 1 > /sys/module/zfs/parameters/zfs_disable_ivset_guid_check
+# zfs send -Rw test/crypt1@snap1 | zfs receive test/newcrypt1
+# zfs send -I test/crypt1@snap1 test/crypt1@snap5 | zfs receive test/newcrypt1
+# zfs destroy -R test/crypt1
+# echo 0 > /sys/module/zfs/parameters/zfs_disable_ivset_guid_check
+</pre>
+
+The errata will be cleared upon reimporting the pool and the alert will
+only be shown again if another dataset is found with the errata. To
+ensure that all datasets are fixed, perform a <b>zpool scrub</b>, and
```

Comment:
Do we really need to do a scrub? It looks like the errata is detected when we open the dataset, so `zfs list -t all` should be sufficient?

Reply:
Thats a good point, thanks.